Upgrade Carbon for IBM.com and Angular

### Description

I'm running Angular 9 and Carbon for IBM.com web components v1.25.0.
I'm trying to upgrade them to fix vulnerabilities found in these components.
Another reason is that IBM CIO pipeline requires newer Angular versions.
I'm stuck, I can't migrate to a more recent Angular and Carbon for IBM.com web components versions.
When I try to migrate Carbon for IBM.com web components v1.26.0 I get the following errors:
```
ERROR in node_modules/@carbon/ibmdotcom-web-components/es/components/image/image.d.ts:329:16 - error TS1005: ')' expected.

329 } & ((abstract new (...args: any[]) => {
                   ~~~
node_modules/@carbon/ibmdotcom-web-components/es/components/image/image.d.ts:329:21 - error TS1109: Expression expected.

329 } & ((abstract new (...args: any[]) => {
                        ~~~
node_modules/@carbon/ibmdotcom-web-components/es/components/image/image.d.ts:329:34 - error TS1011: An element access expression should take an argument.

329 } & ((abstract new (...args: any[]) => {
```

If I try to upgrade to Angular 10, I get:
```
node_modules/@carbon/ibmdotcom-web-components/es/components/masthead/masthead-composite.d.ts:112:61 - error TS1005: ',' expected.

112     closest<K_3 extends keyof SVGElementTagNameMap>(selector: K_3): SVGElementTagNameMap[K_3] | null;
                                                                ~
node_modules/@carbon/ibmdotcom-web-components/es/components/masthead/masthead-composite.d.ts:112:67 - error TS1005: ';' expected.

112     closest<K_3 extends keyof SVGElementTagNameMap>(selector: K_3): SVGElementTagNameMap[K_3] | null;
```

### Component(s) impacted

All Carbon for IBM.com web components.

### Browser

Firefox

### Carbon for IBM.com version

v1.25.0

### Severity

Severity 1 = The design is broken in a critical way that blocks users from completing tasks or damages the brand. Affects major functionality, no workaround.

### Application/website

https://w3east-limited-use.cpc.ibm.com/isc/esd/dswdown/home

### Package

@carbon/ibmdotcom-web-components, @carbon/web-components

### CodeSandbox example

https://github.com/igaldino/stackblitz-carbon-for-ibm-dotcom

### Steps to reproduce the issue (if applicable)

_No response_

### Release date (if applicable)

_No response_

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/blob/main/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues) for duplicate issues

--- a/packages/carbon-web-components/src/components/input/input.ts
+++ b/packages/carbon-web-components/src/components/input/input.ts
@@ -50,6 +50,12 @@ export default class BXInput extends ValidityMixin(FormMixin(LitElement)) {
   protected _value = '';
 
   /**
+   * Set initial value of input
+   */
+  @property({ reflect: true })
+  value = '';
+
+  /**
    * Handles `oninput` event on the `<input>`.
    *
    * @param event The event.
@@ -197,32 +203,6 @@ export default class BXInput extends ValidityMixin(FormMixin(LitElement)) {
    */
   @property({ attribute: 'validity-message' })
   validityMessage = '';
-
-  /**
-   * The value of the input.
-   */
-  @property({ reflect: true })
-  get value() {
-    // FIXME: Figure out how to deal with TS2611
-    // once we have the input we can directly query for the value
-    if (this._input) {
-      return this._input.value;
-    }
-    // but before then _value will work fine
-    return this._value;
-  }
-
-  set value(value) {
-    const oldValue = this._value;
-    this._value = value;
-    // make sure that lit-element updates the right properties
-    this.requestUpdate('value', oldValue);
-    // we set the value directly on the input (when available)
-    // so that programatic manipulation updates the UI correctly
-    if (this._input) {
-      this._input.value = value;
-    }
-  }
 
   createRenderRoot() {
     return this.attachShadow({

--- a/packages/web-components/src/components/footer/combo-box.ts
+++ b/packages/web-components/src/components/footer/combo-box.ts
@@ -12,7 +12,7 @@ import { TemplateResult } from 'lit-html';
 import { html, property, query } from 'lit-element';
 import BXComboBoxItem from '../../internal/vendor/@carbon/web-components/components/combo-box/combo-box-item.js';
 import Close16 from '../../internal/vendor/@carbon/web-components/icons/close/16.js';
-import ddsSettings from '@carbon/ibmdotcom-utilities/es/utilities/settings/settings.js';
+import ddsSettings from '../../internal/vendor/@carbon/ibmdotcom-utilities/utilities/settings/settings';
 import { findIndex, forEach } from '../../globals/internal/collection-helpers';
 import DDSDropdown, { DROPDOWN_KEYBOARD_ACTION } from './dropdown';
 import { carbonElement as customElement } from '../../internal/vendor/@carbon/web-components/globals/decorators/carbon-element.js';
@@ -58,7 +58,7 @@ class DDSComboBox extends DDSDropdown {
    * The `<input>` for filtering.
    */
   @query('input')
-  private _filterInputNode!: HTMLInputElement;
+  protected _filterInputNode!: HTMLInputElement;
 
   /**
    * @param item A combo box item.
@@ -226,7 +226,8 @@ class DDSComboBox extends DDSDropdown {
     return true;
   }
 
-  updated() {
+  updated(changedProperties) {
+    super.updated(changedProperties);
     const { _listBoxNode: listBoxNode } = this;
     if (listBoxNode) {
       listBoxNode.classList.add(`${prefix}--combo-box`);

--- a/packages/web-components/src/components/footer/language-selector-desktop.ts
+++ b/packages/web-components/src/components/footer/language-selector-desktop.ts
@@ -25,7 +25,6 @@ const { stablePrefix: ddsPrefix } = ddsSettings;
  * The API for language selection is still subject to change.
  *
  * @element dds-language-selector-desktop
- * @internal
  */
 @customElement(`${ddsPrefix}-language-selector-desktop`)
 class DDSLanguageSelectorDesktop extends HostListenerMixin(DDSComboBox) {
@@ -39,7 +38,7 @@ class DDSLanguageSelectorDesktop extends HostListenerMixin(DDSComboBox) {
    * The `<input>` for filtering.
    */
   @query('input')
-  private _filterInputNode!: HTMLInputElement;
+  protected _filterInputNode!: HTMLInputElement;
 
   /**
    * Reverts input value to last chosen valid language.
@@ -134,9 +133,8 @@ class DDSLanguageSelectorDesktop extends HostListenerMixin(DDSComboBox) {
   @property({ reflect: true })
   slot = 'language-selector';
 
-  // @ts-ignore
   updated(changedProperties) {
-    super.updated();
+    super.updated(changedProperties);
     if (changedProperties.has('value')) {
       this._lastValidLang = this.value;
     }

--- a/packages/web-components/src/components/footer/language-selector-mobile.ts
+++ b/packages/web-components/src/components/footer/language-selector-mobile.ts
@@ -20,7 +20,6 @@ const { stablePrefix: ddsPrefix } = ddsSettings;
  * The API for language selection is still subject to change.
  *
  * @element dds-language-selector-mobile
- * @internal
  */
 @customElement(`${ddsPrefix}-language-selector-mobile`)
 class DDSLanguageSelectorMobile extends BXSelect {

